src/main/java/com/revature/sfadapter/services/SFAuthService.java

```diff
@@ -1,86 +1,93 @@
 import org.springframework.util.MultiValueMap;
-import org.springframework.util.SystemPropertyUtils;
-import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.client.RestTemplate;
 
-import javax.print.attribute.standard.Media;
-import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 import java.net.URLEncoder;
-import java.util.*;
+import java.util.Arrays;
+import java.util.Map;
 
 @Service
 public class SFAuthService {
 
-	private DaoService<SFWSAccessObject, String> tokenService;
+	private static final String clientId;
+	private static final String clientSecret;
+	private static final String authorizationUrl;
+	private static final String myCallbackUrl;
+
+	static{
+		Map<String, String> env = System.getenv();
+		clientId = env.get("CONSUMER_KEY");
+		clientSecret = env.get("CONSUMER_SECRET");
+		authorizationUrl = env.get("SF_AUTH_SERVER");		// *.saleforce.com/
+		myCallbackUrl = env.get("CALLBACK_URL");
+	}
 
 	private static final Logger LOGGER = Logger.getLogger(SFAuthService.class);
-	@Autowired
-	public void setTokenService(DaoService<SFWSAccessObject, String> tokenService){
-		Assert.notNull(tokenService);
-		this.tokenService = tokenService;
+
+	private String buildAuthorizationUrl(String endpoint, String clientRedirect){
+
+		//StringBuilder url = new StringBuilder(String.format("%s?response_type=code", authorizationUrl));
+		StringBuilder url = new StringBuilder(authorizationUrl.concat(endpoint));
+		url.append("?response_type=code");
+		url.append(String.format("&client_id=%s", clientId));
+		url.append(String.format("&redirect_uri=%s", URLEncoder.encode(myCallbackUrl)));
+		url.append(String.format("&state=%s", URLEncoder.encode(clientRedirect)));
+
+		return url.toString();
 	}
 
-	private String getRequestUrl(String clientRedirect){
-		String clientId = System.getenv("SF_CONSUMER_KEY");
+	private MultiValueMap<String, String> buildAccessRequestFormData(String accessCode){
+		MultiValueMap<String, String> data = new LinkedMultiValueMap<>();
 
-		String url = "https://test.salesforce.com/services/oauth2/authorize?response_type=code";
-		url += "&client_id=" + clientId;
-		url += "&redirect_uri=" + URLEncoder.encode("https://sf.aduet.tech/services/authenticate");
-		url+= "&state=" + URLEncoder.encode(clientRedirect);
-		url += "&prompt=login";
+		data.add("grant_type", "authorization_code");
+		data.add("code", accessCode);
+		data.add("client_id", clientId);
+		data.add("client_secret", clientSecret);
+		data.add("redirect_uri",myCallbackUrl);
 
-		return url;
+		return data;
 	}
 
-	public void sendForAuth(HttpServletResponse response, String clientRedirect) throws IOException{
-		String url = getRequestUrl(clientRedirect);
-		response.sendRedirect(url);
+	public String getAuthUrl(String clientRedirect) throws IOException{
+		LOGGER.debug("Building url for sales force request");
+		String authUrl = buildAuthorizationUrl("/services/oauth2/authorize", clientRedirect);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "authUrl" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revatureops.com/coding_rules#rule_key=squid%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "authUrl". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revatureops.com/coding_rules#rule_key=squid%3AS1854)